File: 프로그래머스/2/12924. 숫자의 표현/숫자의 표현.py
def solution(n):
    answer = 0
    nlist = [i for i in range(n+1)]
    print(nlist)
    
    for i in range(1,n) :
        sumnum = 0 
        for j in range(i,n) :
            sumnum+= j
            if sumnum == n :
                answer+= 1
                break
            elif sumnum > n :
                break
    
    return answer+1

File: 프로그래머스/2/12924. 숫자의 표현/숫자의 표현.java
class Solution {
    public int solution(int n) {
        int answer = 0;
        
        int start = 1; 
        int end = 1;
        int checkval = 1;
        while (end<= (n))
        {
            if (checkval <n)
            {
                //모자르다
                end +=1;
                checkval += end;
            } else if (checkval == n )
            {
                answer +=1;  
                end +=1;
                checkval += end;
            }  else {
                //값이 크다
                checkval -= start;
                start +=1;
            }
            
        }
        return answer;
    }
}